core/src/main/java/hivemall/utils/io/BoundedInputStream.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package hivemall.utils.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Input stream which only supplies bytes up to a certain length
+ * Implementation is based on BoundedInputStream in Apache Commons IO
+ * https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/input/BoundedInputStream.html
+ */
+public final class BoundedInputStream extends InputStream {
```

Comment:
`BoundedInputStream` returns at max N bytes. It's not what we want.

Should throw an exception when reached max bytes as seen in this [LimitedInputStream](https://github.com/apache/commons-fileupload/blob/master/src/main/java/org/apache/commons/fileupload/util/LimitedInputStream.java).

[Fail-fast](https://en.wikipedia.org/wiki/Fail-fast) behavior is expected instead of malformed stream cause exception on GzipInpuStream or Dictionary reader.